src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.errors.ErrorMessages;
+import core.basesyntax.model.Fruit;
+import java.util.List;
+
+public class FruitDaoImpl implements FruitDao {
+    @Override
+    public synchronized void add(Fruit fruit) {
+        Storage.addFruit(fruit);
+    }
+
+    @Override
+    public synchronized Fruit getFruitByName(String fruitName) {
+        return Storage.getFruits().stream()
+                .filter(f -> f.getName().equals(fruitName))
```

Comment:
please don't use short one letter names even in stream chains - reduces readability

Reply:
Ok